lib/milc_interface.cpp

```diff
@@ -1137,14 +1153,72 @@ void qudaMultishiftInvert(int external_precision, int quda_precision, int num_of
   qudamilc_called<false>(__func__, verbosity);
 } // qudaMultiShiftInvert
 
+// Wrapper function for qudaInvertDeflatable to maintain backward compatibility with old(er) MILC
 void qudaInvert(int external_precision, int quda_precision, double mass, QudaInvertArgs_t inv_args,
                 double target_residual, double target_fermilab_residual, const void *const fatlink,
                 const void *const longlink, void *source, void *solution, double *const final_residual,
                 double *const final_fermilab_residual, int *num_iters)
 {
+  
+  // If this function is called then QUDA is not doing deflation
+  // Create dummy QudaEigensolverArgs_t that requests 0 eigenvalues
+  QudaEigensolverArgs_t eig_args;
+  eig_args.n_ev_deflate=0;
+  eig_args.prec_eigensolver = QUDA_DOUBLE_PRECISION; /* Not used in the undeflated case but needs to be initialized */
+
+  qudaInvertDeflatable(external_precision, quda_precision, mass, inv_args, eig_args,
+                target_residual, target_fermilab_residual, fatlink,
+                longlink, source, solution, final_residual,
+                final_fermilab_residual, num_iters);
+
+} // qudaInvert
+
+
+void qudaInvertDeflatable(int external_precision, int quda_precision, double mass, QudaInvertArgs_t inv_args, QudaEigensolverArgs_t eig_args,
+                double target_residual, double target_fermilab_residual, const void *const fatlink,
+                const void *const longlink, void *source, void *solution, double *const final_residual,
+                double *const final_fermilab_residual, int *num_iters)
+{
   static const QudaVerbosity verbosity = getVerbosity();
   qudamilc_called<true>(__func__, verbosity);
 
+  QudaParity local_parity = inv_args.evenodd;
+  
+  // parameters for the eigensolve/deflation
```

Comment:
For sanity/cleanliness/reuse between this and the msrc variety, this initialization should be moved to a function `setEigensolverParams` in the same spirit of `setInvertParams` and `setGaugeParams`.